src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -840,16 +841,15 @@ class DrawSketchHandlerGenConstraint: public DrawSketchHandler
 
     virtual void activated(ViewProviderSketch *)
     {
-        Gui::Selection().rmvSelectionGate();
+        ongoingSequences = new std::set<int>(); //TODO: make it contain more values
+        _tempOnSequences = new std::set<int>();
+        selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
 
-        // Estimate allowed selections from the first types in allowedSelTypes
-        int allowedSelTypes = 0;
-        for (std::vector< std::vector< SelType > >::const_iterator it = cmd->allowedSelSequences.begin();
-             it != cmd->allowedSelSequences.end(); ++it) {
-            allowedSelTypes = allowedSelTypes | *((*it).begin());
-        }
-        GenericConstraintSelection* selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
-        selFilterGate->setAllowedSelTypes(allowedSelTypes);
+        resetOngoingSequences();
+
+        selSeq = *(new std::vector<SelIdPair>());
```

Comment:
`new std::vector<SelIdPair>()` creates a new instance of std::vector on the heap
`selSeq = *(new std::vector<SelIdPair>())` dereferences the pointer and assigns its content to the existing variable

> Before this point selseq doesn't exist so "selSeq.clear();" will fail.

How can this ever fail? selSeq is a member variable and automatically set in the constructor. I guess you are mixing up C++ with JAVA.